usr.sbin/daemon/daemon.c

```diff
@@ -742,3 +761,21 @@ reopen_log(struct log_params *logparams)
 	logparams->output_fd = outfd;
 }
 
+static _Noreturn void
+daemon_terminate(struct daemon_state * state, int exitcode)
+{
+	if (state == NULL) {
+		exit(1);
```

Comment:
why aren't you passing on the exit code here?

Reply:
this is abnormal behavior, state should never be null, so we exit with 1 if that happens